presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainOrcPredicate.java

```diff
@@ -75,9 +96,149 @@ public boolean matches(long numberOfRows, Map<Integer, ColumnStatistics> statist
+                    numberOfBloomFiltersChecked++;
+                    if (matched) {
+                        // bloom filter is matched here return true so we select this stripe as it likely contains data which we need to read
+                        return true;
+                    }
+                }
+            }
+        }
+
+        // did we read any bloom filter?
+        if (numberOfBloomFiltersChecked < 1) {
+            // no bloom filters read, seems like a query that was run only on partitioned columns
+            return true;
+        }
+
+        // none of the bloomfilters caused a "hit" meaning we should not read
+        log.debug("Not reading, didn't match any of the bloomfilters (" + numberOfBloomFiltersChecked + "x), data is not here");
+        return false;
+    }
+
+    @VisibleForTesting
+    public static Collection<Object> predicateValuesFromDomain(Domain domain)
+    {
+        ValueSet values = domain.getValues();
+        if (values instanceof EquatableValueSet) {
+            EquatableValueSet eqValues = (EquatableValueSet) values;
+            if (eqValues.isWhiteList()) {
+                // we can only work with values we know, not excluded blacklists because other rows might contain the data we need
+                return ImmutableSet.of(values.getDiscreteValues().getValues());
+            }
+        }
+        else if (values instanceof SortedRangeSet) {
+            SortedRangeSet sortedRangeSet = (SortedRangeSet) values;
+            // sorted range set is used for integer comparison (e.g. id = 123 ) where min and max is the same value
+            if (sortedRangeSet.isSingleValue()) {
+                return ImmutableSet.of(sortedRangeSet.getSingleValue());
+            }
+        }
+        return ImmutableSet.of();
+    }
+
+    // checks whether a value part of the effective predicate is likely to be part of this bloomfilter
+    @VisibleForTesting
+    public static boolean checkInBloomFilter(BloomFilter bloomfilter, Object predicateValue)
+    {
+        if (predicateValue instanceof Long) {
+            return bloomfilter.testLong((Long) predicateValue);
+        }
+        else if (predicateValue instanceof Double) {
+            return bloomfilter.testDouble((Double) predicateValue);
+        }
+        else if (predicateValue instanceof Slice) {
+            Slice slice = (Slice) predicateValue;
+            return bloomfilter.testString(slice.toStringUtf8());
+        }
+        else if (predicateValue instanceof String ||
+                predicateValue instanceof BigDecimal) {
+            return bloomfilter.testString(predicateValue.toString());
+        }
+        else if (predicateValue instanceof Timestamp) {
+            return bloomfilter.testLong(((Timestamp) predicateValue).getTime());
+        }
+        log.warn("Bloom filter check not supported for type " + predicateValue.getClass() + " " + predicateValue.toString());
```

Comment:
Remove this, too. If a new type is added that isn't supported here, it will flood the logs. It's also not something the end user can do anything about.